### PR TITLE
Don't remove output files when a test fails

### LINKED ISSUE #1726: Output from InputFiles tests should not be removed if the test fails
I am seeing rare failures that appear to be due to some kind of H5 file corruption, and this is difficult to diagnose when the possibly corrupt files are immediately removed by the test system.

We should make sure that rerunning a failed test still works after changing this.

### PR DESCRIPTION
**Pull request by wthrowe**
Fixes #1726

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
@nilsleiffischer please look at this when you have a moment.

File: tools/CleanOutput.py [hunk 1]
```diff
@@ -84,13 +84,23 @@ def parse_args():
...
     return parser.parse_args()
 
 
 if __name__ == "__main__":
     args = parse_args()
 
-    # Set the log level
-    logging.basicConfig(level=logging.WARNING - args.verbose * 10)
+    if args.force:
+        logging.basicConfig(level=1000)
```
Comment (nilsvu) [inline tools/CleanOutput.py]: Can we add this feature in a way that doesn’t disable logging? How about you just pass the `force` flag to the `clean_output` function and  use it in line 56 by replacing the `else` with `elif not force`? I think that‘ll do what you want, and still provide useful output.

You can then also add the `-v` flag to where you call the script, so the log will report „Removed file ...“ messages.

(As a side note: The PEP8 guidelines strongly discourage bare `except` clauses (E722). We should probably add a linter like autopep8 to our CI ASAP)

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by wthrowe** [comment]
Squashed because it was basically a complete rewrite of the patch to the python script.

As a mostly unrelated question, is there a reason this warns if a test doesn't produce any output?  There's nothing wrong with that situation and producing extra warnings is bad.  If the list is actually missing you're going to find out because the test explodes the second time you run it.

**Review by nilsvu — APPROVED** [review]
LGTM 👍

About your question: Input files can specify an empty list of expected outputs to explicitly state they don‘t produce any.
I think when I wrote the script my intention was to warn of a „hidden default“: When a user doesn‘t add the list to the input file because they don’t know about the feature, no outputs are checked even though the executable may have some. If you disagree, please open an issue.

**Review by kidder — APPROVED** [review]

**Pull request merged**